code/modules/admin/verbs/adminpanelweapons.dm

```diff
@@ -0,0 +1,125 @@
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 1, mob.loc, point_defense), hiteta SECONDS)
+				if(point_defense == "Yes")
+					var/spoolup = hiteta - 4
+					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(shipwide_ai_announcement), "ATTENTION: TRACKING TARGET, SPOOLING UP POINT DEFENSE. ATTEMPTING TO INTERCEPT." , MAIN_AI_SYSTEM, 'sound/effects/supercapacitors_charging.ogg'), spoolup SECONDS)
+
+			if(exactplace == "Random")
+				if(salvo == "Salvo")
+					shipwide_ai_announcement("DANGER: MISSILE SALVO DETECTED, BRACE, BRACE, BRACE. SALVO SIZE: [quantity] , ESTIMATED TIME: [hiteta] SECONDS." , MAIN_AI_SYSTEM, 'sound/effects/missile_warning.ogg')
+					for(currentshot = 1; currentshot <= quantity; currentshot++ )
+						while(picked_atom == null)
+							picked_area = pick(GLOB.ship_areas)
+							for(var/turf/my_turf in picked_area)
+								if(isturf(my_turf))
+									turfs_of_area += my_turf
+							if(turfs_of_area.len > 1)
+								picked_atom = pick(turfs_of_area)
+								targets += picked_atom
+						picked_atom = null
+					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 1, targets, point_defense, salvo), hiteta SECONDS)
+					if(point_defense == "Yes")
+						var/spoolup = hiteta - 4
+						addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(shipwide_ai_announcement), "ATTENTION: TRACKING TARGETS, SPOOLING UP POINT DEFENSE. ATTEMPTING TO INTERCEPT." , MAIN_AI_SYSTEM, 'sound/effects/supercapacitors_charging.ogg'), spoolup SECONDS)
+				else
+					shipwide_ai_announcement("DANGER: MISSILE WARNING. LAUNCH DETECTED, BRACE, BRACE, BRACE. ESTIMATED TIME: [hiteta] SECONDS.", MAIN_AI_SYSTEM, 'sound/effects/missile_warning.ogg')
+					while(picked_atom == null)
+						picked_area = pick(GLOB.ship_areas)
+						for(var/turf/my_turf in picked_area)
+							if(isturf(my_turf))
+								turfs_of_area += my_turf
+
+						if(turfs_of_area.len > 1)
+							picked_atom = pick(turfs_of_area)
+
+					addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 1, picked_atom, point_defense), hiteta SECONDS)
+					if(point_defense == "Yes")
+						var/spoolup = hiteta - 4
+						addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(shipwide_ai_announcement), "ATTENTION: TRACKING TARGET, SPOOLING UP POINT DEFENSE. ATTEMPTING TO INTERCEPT." , MAIN_AI_SYSTEM, 'sound/effects/supercapacitors_charging.ogg'), spoolup SECONDS)
+
+		if("Railgun")
+			if(exactplace == "Where I am")
+				shipwide_ai_announcement("DANGER: RAILGUN EMISSIONS DETECTED, INCOMING SHOT. BRACE, BRACE, BRACE. ESTIMATED TIME: [hiteta] SECONDS." , MAIN_AI_SYSTEM, 'sound/effects/missile_warning.ogg')
+				while(picked_atom == null)
+					picked_area = pick(GLOB.ship_areas)
+
+					for(var/turf/my_turf in picked_area)
+						if(isturf(my_turf))
+							turfs_of_area += my_turf
+
+					if(turfs_of_area.len > 1)
+						picked_atom = pick(turfs_of_area)
+
+				addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(weaponhits), 2, mob.loc, point_defense), hiteta SECONDS)
```

Comment:
In case length is less than 1 (should be an >=, actually, whoops), picked_atom stays as null, repeats this piece of code to make sure it actually picks somethin'

Reply:
Github may or may not be swhwoing the right part of the code, and It doesn't show on VSC for no good reason, though I may see what yo'ure talking about. In the Railgun section, aye?

Reply:
When it appears on VSC it's luckily much more on point, and it appeared for me there so yeah, I got it now!